### PR TITLE
docs: WIP gh actions job to scrape docs for search

### PR DESCRIPTION
**Pull request by johanandren**
Along with this I have added two new gh actions secrets for algolia write API key and algolia project id, and also added the darrenjennings/algolia-docsearch-action@0.2.0 action to our allow-list

### CODE REVIEW

File: .algolia-search-index-config.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  "custom_settings": {
+    "attributesForFaceting": [
+      "language"
+    ]
+  },
+  "conversation_id": [
+    "371144203"
+  ],
+  "nb_hits": 5936
+}
```
Comment (johanandren) [inline .algolia-search-index-config.json:44]: This is the existing OSS scraper config, 🤞 that it works to just pick and use

**Review by octonato — APPROVED** [review]
LGTM

But what is the goal exactly? To scrape out some tagging to make it easier for agolia to index the pages?

**Comment by johanandren** [comment]
Algolia had set up and was doing indexing/scraping for us for free, now that we are not OSS anymore we need to do that scraping and updating of the search index ourselves.

**Comment by johanandren** [comment]
Let's see if this works.

**Comment by johanandren** [comment]
Didn't work, but now that the workflow exists in main, I can test changes without merging. Continued in #31873

**Pull request merged**